shared/modules/selectors/networks.ts

```diff
@@ -43,6 +45,81 @@ export function getSelectedNetworkClientId(
 ) {
   return state.metamask.selectedNetworkClientId;
 }
+/**
+ * Type to extend InternalNetworkState with multichain configurations.
+ */
+type ExtendedInternalNetworkState = InternalNetworkState & {
+  multichainNetworkConfigurationsByChainId: Record<
+    string,
+    NetworkConfiguration
+  >;
+};
+
+/**
+ * State type that includes both standard and multichain network configurations.
+ */
+export type ConsolidatedNetworkConfigurationsState = {
+  metamask: Pick<
+    ExtendedInternalNetworkState,
+    | 'networkConfigurationsByChainId'
+    | 'multichainNetworkConfigurationsByChainId'
+  >;
+};
+
+/**
+ * Combines and returns network configurations for all chains (EVM and not).
+ *
+ * @param state - Redux state.
+ * @returns A consolidated object containing all available network configurations.
+ */
+export const getAllNetworkConfigurationsByCaipChainId = createDeepEqualSelector(
+  (state: ConsolidatedNetworkConfigurationsState) =>
+    state.metamask.networkConfigurationsByChainId,
+  (state: ConsolidatedNetworkConfigurationsState) =>
+    state.metamask.multichainNetworkConfigurationsByChainId,
+  (state: {
+    metamask: { internalAccounts: AccountsControllerState['internalAccounts'] };
+  }) => state.metamask.internalAccounts,
+  (
+    networkConfigurationsByChainId,
+    multichainNetworkConfigurationsByChainId,
+    internalAccounts,
+  ) => {
+    const caipFormattedEvmNetworkConfigurations: Record<
+      string,
+      NetworkConfiguration
+    > = {};
+
+    Object.entries(networkConfigurationsByChainId).forEach(
+      ([chainId, network]) => {
+        const caipChainId = `eip155:${hexToDecimal(chainId)}`;
+        caipFormattedEvmNetworkConfigurations[caipChainId] = network;
+      },
+    );
+
+    Object.entries(multichainNetworkConfigurationsByChainId).forEach(
```

Comment:
```suggestion
// For now we need to filter out networkConfigurations/scopes without accounts because
// the `endowment:caip25` caveat validator will throw if there are no supported accounts for the given scope
    Object.entries(multichainNetworkConfigurationsByChainId).forEach(
```